doc/warm-reboot/NSF_Manager_HLD.md

```diff
@@ -0,0 +1,280 @@
+# NSF Manager HLD
+
+
+## Table of Content
+
+
+- [Revision](#revision)
+- [Scope](#scope)
+- [Definitions/Abbreviations](#definitions-abbreviations)
+- [Overview](#overview)
+- [Requirements](#requirements)
+- [Architecture Design](#architecture-design)
+- [High-Level Design](#high-level-design)
+  * [NSF Details Registration](#nsf-details-registration)
+  * [Shutdown Orchestration](#shutdown-orchestration)
+    + [Phase 1: Freeze Components & Wait for Switch Quiescence](#phase-1--freeze-components---wait-for-switch-quiescence)
+    + [Phase 2: State Verification (Optional)](#phase-2--state-verification--optional-)
+    + [Phase 3: Trigger Checkpointing](#phase-3--trigger-checkpointing)
+    + [Phase 4: Prepare and Perform Reboot](#phase-4--prepare-and-perform-reboot)
+    + [Application Shutdown Optimization](#application-shutdown-optimization)
+  * [Reconciliation Monitoring](#reconciliation-monitoring)
+  * [Component Warmboot States](#component-warmboot-states)
+- [SAI API](#sai-api)
+- [Configuration and management](#configuration-and-management)
+- [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+- [Restrictions/Limitations](#restrictions-limitations)
+- [Testing Requirements/Design](#testing-requirements-design)
+- [Open/Action items - if any](#open-action-items---if-any)
+
```

Comment:
Couple of sections can be added. one use cases covered ( LACP.. etc). Also sections which NPU types are tested or supported after testing.

Reply:
Added NPU types detail in Testing Requirements/Design section.